src/main/java/seedu/address/model/person/Id.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Student's id in the record.
+ * Id is the last 4 digits of the NRIC (e.g. 123A).
+ * Guarantees: immutable; is valid as declared in {@link #isValidId(String)}
+ */
+public class Id {
+
+    public static final String MESSAGE_CONSTRAINTS = "Id should only contain 3 digits and 1 character";
+    public static final String VALIDATION_REGEX = "\\d{3}[a-zA-Z]";
+    public final String value;
```

Comment:
Should this be private such that the Id field cannot be accessed outside of the class?

Reply:
Hmm it's actually accessed outside the class. Like same for the other fields

Reply:
Okay, since value if final too so it can't be changed. If you want to, could implement a getter method.